src/type-mapper.cc

```diff
@@ -24,7 +24,13 @@ TypeMapper::infer_type(const Local<Value>& value)
         return CASS_VALUE_TYPE_BOOLEAN;
     }
     else if (value -> IsObject()) {
-        return CASS_VALUE_TYPE_MAP;
+        // If the value has a low/high key it's a bigint, otherwise it's a map
```

Comment:
I don't think this makes sense to do this kind of introspection at the inference layer since it could cause confusion if someone happens to want to store a map with those two fields. It seems better to me if the caller has to use the explicit type hints when using BIGINT.